src/functional/iterator/embedded.py

```diff
@@ -391,7 +419,7 @@ def execute_shift(
         else:
             raise AssertionError()
         return new_pos
-    elif isinstance(offset_implementation, NeighborTableOffsetProvider):
+    elif isinstance(offset_implementation, Connectivity):
```

Comment:
Ok, let's do 
```suggestion
    else:
        assert isinstance(offset_implementation, Connectivity):
```
as Till suggested. 
Also then remove the AssertionError in line 434.

@tehrengruber I don't get why it is not sufficient here. Connectivity is not well defined, e.g. the `tbl` should not be part of the public API, but as far as I can see it should work here, no?